validator/slashing-protection/remote.go

```diff
@@ -0,0 +1,151 @@
+	}
+
+	md := make(metadata.MD)
+	for _, hdr := range strings.Split(config.GrpcHeadersFlag, ",") {
+		if hdr != "" {
+			ss := strings.Split(hdr, "=")
+			if len(ss) != 2 {
+				log.Warnf("Incorrect gRPC header flag format. Skipping %v", hdr)
+				continue
+			}
+			md.Set(ss[0], ss[1])
+		}
+	}
+
+	opts := []grpc.DialOption{
+		dialOpt,
+		grpc.WithDefaultCallOptions(
+			grpc_retry.WithMax(config.GrpcRetriesFlag),
+			grpc_retry.WithBackoff(grpc_retry.BackoffLinear(config.GrpcRetryDelay)),
+			grpc.Header(&md),
+		),
+		grpc.WithStatsHandler(&ocgrpc.ClientHandler{}),
+		grpc.WithStreamInterceptor(middleware.ChainStreamClient(
+			grpc_opentracing.StreamClientInterceptor(),
+			grpc_prometheus.StreamClientInterceptor,
+			grpc_retry.StreamClientInterceptor(),
+		)),
+		grpc.WithUnaryInterceptor(middleware.ChainUnaryClient(
+			grpc_opentracing.UnaryClientInterceptor(),
+			grpc_prometheus.UnaryClientInterceptor,
+			grpc_retry.UnaryClientInterceptor(),
+			grpcutils.LogGRPCRequests,
+		)),
+	}
+	return &RemoteProtector{
+		ctx:             ctx,
+		slasherEndpoint: config.SlasherEndpoint,
+		opts:            opts,
+	}, nil
+}
+
+// Start the remote protector service.
+func (rp *RemoteProtector) Start() {
+	conn, err := grpc.DialContext(rp.ctx, rp.slasherEndpoint, rp.opts...)
+	if err != nil {
+		log.Errorf("Could not dial slasher endpoint: %s", rp.slasherEndpoint)
+		return
+	}
+	rp.slasherClient = slashpb.NewSlasherClient(conn)
+	log.Debug("Successfully started slasher gRPC connection")
+}
+
+// Stop the remote protector service.
+func (rp *RemoteProtector) Stop() error {
+	log.Info("Stopping remote slashing protector")
+	if rp.conn != nil {
+		return rp.conn.Close()
+	}
+	return nil
+}
+
+// Status checks if the connection to slasher server is ready,
+// returns error otherwise.
+func (rp *RemoteProtector) Status() error {
+	if rp.conn == nil {
+		return errors.New("no connection to slasher RPC")
+	}
+	if rp.conn.GetState() != connectivity.Ready {
+		return fmt.Errorf(
+			"cannot connect to slasher server at: %v connection status: %v",
+			rp.slasherEndpoint,
+			rp.conn.GetState(),
+		)
+	}
+	return nil
+}
+
+// CheckBlockSafety this function is part of slashing protection for block proposals it performs
+// validation without db update. To be used before the block is signed.
+func (rp *RemoteProtector) IsSlashableBlock(
+	ctx context.Context, block *ethpb.SignedBeaconBlock, pubKey [48]byte, domain *ethpb.DomainResponse,
+) (bool, error) {
+	signedHeader, err := blockutil.SignedBeaconBlockHeaderFromBlock(block)
+	if err != nil {
+		return false, errors.Wrap(err, "could not extract signed header from block")
+	}
+	resp, err := rp.slasherClient.IsSlashableBlock(ctx, signedHeader)
+	if err != nil {
+		return false, errors.Wrap(err, "remote slashing block protection returned an error")
+	}
+	return resp != nil && resp.ProposerSlashing != nil, nil
+}
```

Comment:
Please put these API methods into other go files.

remote_block.go
remote_attestation.go

That makes for easier navigation, smaller test files, etc.